### PR TITLE
Support for landmarks in g2o files

### PR DESCRIPTION
**Pull request by varunagrawal**
Fix for #242.

### CODE REVIEW

**Comment by varunagrawal** [comment]
So I only need to create unit tests for this via a sample dataset which also has landmarks. I believe the right approach is to just make my own simple dataset with a robot moving in a circle or a straight line.

File: gtsam/slam/dataset.h [hunk 1]
```diff
@@ -76,14 +76,24 @@ enum KernelFunctionType {
...
+typedef std::pair<Key, Point2> IndexedLandmark;
 typedef std::pair<std::pair<Key, Key>, Pose2> IndexedEdge;
 
 /**
  * Parse TORO/G2O vertex "id x y yaw"
  * @param is input stream
  * @param tag string parsed from input stream, will only parse if vertex type
  */
-GTSAM_EXPORT boost::optional<IndexedPose> parseVertex(std::istream& is,
+GTSAM_EXPORT boost::optional<IndexedPose> parseVertexPose(std::istream& is,
```
Comment (dellaert) [inline gtsam/slam/dataset.h:87]: I see no reason for an API breaking name change, and if it were to be implemented we need a deprecated section with the old name.

Comment (varunagrawal) [reply gtsam/slam/dataset.h:87]: Sounds good. I'll use the compiler definition from #422 if that's okay?

Comment (dellaert) [reply gtsam/slam/dataset.h:97]: I prefer *not* to do the name change

**Comment by varunagrawal** [comment]
It felt more consistent since I added a function named `parseVertexLandmark`. It also avoided ambiguity thanks to the name: you know you are parsing a g2o vertex which represents a pose.

If you still feel the consistency is not worth the extra deprecation effort, just give this message a *thumbs down* and I'll undo it. :slightly_smiling_face:

**Pull request merged**